### ISSUE TITLE
Feature: Advanced Marker supports slot customization elements.

### ISSUE THREAD

**Issue by coderwyd** [inocan-group/vue3-google-map#275]
The specific usage is as follows:
`<AdvancedMarker :options="markerOptions">
    <div>custom content</div>
</AdvancedMarker >`

**Comment by BenJackGill** [comment]
Yeah I agree. This would be great. Without a slot for custom content it's kind of useless for me now.

**Comment by richgcook** [comment]
This would be a great feature request especially with the upcoming deprecation(s).

**Comment by BenJackGill** [comment]
Any movement on this one? I would try and do a pull request myself but I'm not skilled enough to figure it out :/

Until slots are supposed with custom content I am stuck using the `CustomMarker` which gives a console hefty console warning.

**Comment by sk8gear** [comment]
There is a workaround to customize the AdvanceMarker by using "content" inside options. The key is to return a function to HtmlDivElement into the "content"

E.g:
map.vue
`
<script setup>
import { GoogleMap, AdvancedMarker } from 'vue3-google-map'

const pinOptions = { background: '#FBBC04' }

function content(){
 const content = document.createElement("div")
 content.innerHtml = "<h3>Here's the HTML element</h3>"
 return content
}
</script>

<template>
  <GoogleMap
    api-key="YOUR_GOOGLE_MAPS_API_KEY"
    mapId="DEMO_MAP_ID"
    style="width: 100%; height: 500px"
    :center="center"
    :zoom="15"
  >
    <AdvancedMarker :options="{ position: { lat: 40.689247, lng: -74.044502 }, content: content}" :pin-options="pinOptions"/>
  </GoogleMap>
</template>
`

If you want to use a Vue component so we can style it as we want, we can make it like this:

map.vue
`
<script setup>
import mapballon from './mapballon.vue'
import { GoogleMap, AdvancedMarker } from 'vue3-google-map'

const pinOptions = { background: '#FBBC04' }

function content(){
 // here's the code to return our mapballon.vue into HTMLElement
 return document.findElementbyId("mystyle")
}
</script>

<template>
  <GoogleMap
    api-key="YOUR_GOOGLE_MAPS_API_KEY"
    mapId="DEMO_MAP_ID"
    style="width: 100%; height: 500px"
    :center="center"
    :zoom="15"
  >
    <mapballon id="mymapballon" /> 
    <AdvancedMarker :options="{ position: { lat: 40.689247, lng: -74.044502 }, content: content}" :pin-options="pinOptions"/>
  </GoogleMap>
</template>
`

mapballon.vue
`
<script setup>
// your code here
</script>
<template>
 <div class="mystyle">
   <h3>Here's the HTML element</h3>
 </div>
</template>
<style scope>
.mystyle{
  background: white;
}
</style>
`

But it'll be nice if we can have a cleaner way to handle the content.

**Comment by HusamElbashir** [comment]
Released in `v0.22.0`. Shout out to @coderwyd for his contribution!

**Comment by recrit** [comment]
This is causing a regression when using an InfoWindow as shown on https://github.com/inocan-group/vue3-google-map?tab=readme-ov-file#use-with-marker
```
<AdvancedMarker ...>
      <InfoWindow ...>
      </InfoWindow>
</AdvancedMarker>
```

I created the issue #330 for a fix.
